client/src/components/Endpoints/getIcon.tsx

```diff
@@ -0,0 +1,143 @@
+import React from 'react';
+import { Plugin, GPTIcon, AnthropicIcon } from '~/components/svg';
+import { useAuthContext } from '~/hooks';
+import { cn } from '~/utils';
+import { IconProps } from '~/common';
+
+const getIcon: React.FC<IconProps> = (props) => {
+  const { size = 30, isCreatedByUser, button, model = true } = props;
+  // eslint-disable-next-line react-hooks/rules-of-hooks
+  const { user } = useAuthContext();
+
+  if (isCreatedByUser) {
+    const username = user?.name || 'User';
+
+    return (
+      <div
+        title={username}
+        style={{
+          width: size,
+          height: size,
+        }}
+        className={`relative flex items-center justify-center ${props.className || ''}`}
+      >
+        <img
+          className="rounded-sm"
+          src={
+            user?.avatar ||
+            `https://api.dicebear.com/6.x/initials/svg?seed=${username}&fontFamily=Verdana&fontSize=36`
+          }
+          alt="avatar"
+        />
+      </div>
+    );
+  } else {
+    const { endpoint, error } = props;
+    let icon, bg, name;
+
+    switch (endpoint) {
+      case 'azureOpenAI': {
+        const { chatGptLabel } = props;
+        icon = <GPTIcon size={size * 0.7} />;
+        bg = 'linear-gradient(0.375turn, #61bde2, #4389d0)';
+        name = chatGptLabel || 'ChatGPT';
+        break;
+      }
+
+      case 'openAI': {
+        const { chatGptLabel } = props;
```

Comment:
also you are still running in the problem I outlined.

```js
const { chatGptLabel } = props;
...
const { chatGptLabel } = props;
...
const { chatGptLabel } = props;
...
const { chatGptLabel } = props;
```

I know this was here before but now is the opportunity to fix it